test/lit/passes/optimize-instructions.wast

```diff
@@ -0,0 +1,11557 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py
+;; RUN: wasm-opt %s --optimize-instructions -S -o - | filecheck %s
+
+(module
+  (memory 0)
+  (type $0 (func (param i32 i64)))
+ ;; CHECK: (func $f (param $i1 i32) (param $i2 i64)
+ ;; CHECK:  (drop
+ ;; CHECK:   (i32.and
+ ;; CHECK:    (local.get $i1)
+ ;; CHECK:    (i32.const 1)
+ ;; CHECK:   )
+ ;; CHECK:  )
```

Comment:
should each of these be alongside the input? (and shouldn't they be right after it, not before?)

Reply:
Styles vary, but the style we use most in our WebAssembly tests is to have the filecheck assertions for a function immediately above the function, rather than below it. The script splits up the output by function because there is no reliable way to differentiate between different logical test cases within a function. Unfortunately this file has a lot of huge functions, but I have a WIP follow up to break them up a bit. For this PR I just copied each module in the original test file into a new lit test file without any further changes.